### PR TITLE
Setup CI

### PR DESCRIPTION
**Pull request by aryanpatel3**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Run type-check and ESLint
+        run: npx lint-staged
```
Comment (gzcharleszhang) [inline .github/workflows/ci.yml]: did you test if this returns non-zero status code if there are linting errors?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```
Comment (gzcharleszhang) [inline .github/workflows/ci.yml]: why do you need this here?

File: .husky/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+_
```
Comment (gzcharleszhang) [inline .husky/.gitignore:1]: what is this ignoring

File: package.json [hunk 4]
```diff
@@ -5,7 +5,10 @@
   "scripts": {
     "kenson": "next dev",
     "build": "next build",
-    "start": "next start"
+    "start": "next start",
+    "prepare": "husky install",
+    "type-check": "tsc --project tsconfig.json --pretty --noEmit",
+    "lint": "eslint --ext js,jsx,ts,tsx --fix"
```
Comment (gzcharleszhang) [inline package.json]: I thought next.js already comes with eslint support out of the box, why are we configuring eslint ourselves?

File: .husky/.gitignore (hunk 3 above)
Comment (aryanpatel3) [reply .husky/.gitignore:1]: husky creates a folder with name _. That folder contains scripts which run before/after git actions like commit

Comment (gzcharleszhang) [reply .husky/.gitignore:1]: I see, husky is weird lol

File: package.json (hunk 4 above)
Comment (aryanpatel3) [reply package.json]: You're right, I just found out. I followed a tutorial to set this up. I'll make the changes and remove this.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aryanpatel3) [reply .github/workflows/ci.yml]: yes I did

File: .github/workflows/ci.yml (hunk 2 above)
Comment (aryanpatel3) [reply .github/workflows/ci.yml]: that is for caching node_modules. Is that not something we want?

Comment (gzcharleszhang) [reply .github/workflows/ci.yml]: well `key` is already used for cacheing, I'm just wondering why you want `restore-keys`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gzcharleszhang) [reply .github/workflows/ci.yml]: why does this step say `No staged files found.`, does `lint-staged` work in the CI since we're not staging anything?

Comment (aryanpatel3) [reply .github/workflows/ci.yml]: It says that because there were no TS/JS files changed. I'm not sure what you mean by the second question.

Comment (gzcharleszhang) [reply .github/workflows/ci.yml]: From what I understand, `lint-staged` seems to be used before you commit changes, when the changes are staged by git. When you're running CI, is anything staged? I'm not sure if this library is meant to be used in the CI. But if you say it works for catching errors then sure

File: .github/workflows/ci.yml (hunk 2 above)
Comment (aryanpatel3) [reply .github/workflows/ci.yml]: It's so that it'll match with a generic key without the hash in case there was a cache miss using the key

Comment (gzcharleszhang) [reply .github/workflows/ci.yml]: when will the cache be populated with a generic key?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gzcharleszhang) [reply .github/workflows/ci.yml]: I'd be more comfortable if you push a commit with linting errors and seeing the github action fail on that commit

File: tsconfig.json [hunk 5]
```diff
@@ -21,9 +17,14 @@
   "include": [
     "next-env.d.ts",
     "**/*.ts",
-    "**/*.tsx"
-, "components/RadioCard.js", "pages/trainer/index.js", "components/wordPrompt.js", "components/preRap.js", "pages/theme.js", "pages/index.js", "pages/_app.js"  ],
-  "exclude": [
-    "node_modules"
-  ]
+    "**/*.tsx",
+    "components/RadioCard.js",
```
Comment (gzcharleszhang) [inline tsconfig.json]: can we do something like `"**/*.js"`?

**Review by gzcharleszhang — APPROVED** [review]
just one comment, otherwise looks good

**Pull request merged**